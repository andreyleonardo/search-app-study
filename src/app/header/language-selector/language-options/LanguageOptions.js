import React from 'react';
import PropTypes from 'prop-types';
import { map } from 'lodash';
import i18n from '../../../../i18n';

import './LanguageOptions.scss';

const LanguageOptions = ({ onClick }) => (
  <div className="language_options row center-xs">
    { map(i18n.availableLanguages, (language, displayLanguage) => (
      <div key={language} className="col-xs">
        <button id={displayLanguage} type="button" className="primary" onClick={() => onClick(language)}>{displayLanguage}</button>
      </div>
    ))}
  </div>
);

LanguageOptions.propTypes = {
  onClick: PropTypes.func.isRequired
};

export default LanguageOptions;
