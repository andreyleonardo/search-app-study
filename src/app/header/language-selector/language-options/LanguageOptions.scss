@import "src/stylesheets/css/_colors";

.language_options {
  display: flex;
  background-color: $primary;
}
