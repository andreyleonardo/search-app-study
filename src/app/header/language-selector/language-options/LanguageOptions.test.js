import React from 'react';
import LanguageOptions from './LanguageOptions';

describe('Language Options', () => {
  let languageOptions;
  let props;

  beforeEach(() => {
    props = {
      onClick: jest.fn()
    };
    languageOptions = shallow(<LanguageOptions {...props} />);
  });

  it('renders without crash', () => {
    expect(languageOptions).toMatchSnapshot();
  });

  it('calls onClick when an option is clicked', () => {
    languageOptions.find('#en').simulate('click');

    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
