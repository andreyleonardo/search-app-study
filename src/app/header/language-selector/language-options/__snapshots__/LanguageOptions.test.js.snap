// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Options renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LanguageOptions
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs"
        >
          <button
            className="primary"
            id="pt"
            onClick={[Function]}
            type="button"
          >
            pt
          </button>
        </div>,
        <div
          className="col-xs"
        >
          <button
            className="primary"
            id="en"
            onClick={[Function]}
            type="button"
          >
            en
          </button>
        </div>,
      ],
      "className": "language_selector__languages row center-xs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "pt-BR",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="primary"
            id="pt"
            onClick={[Function]}
            type="button"
          >
            pt
          </button>,
          "className": "col-xs",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "pt",
            "className": "primary",
            "id": "pt",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "pt",
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "en-US",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="primary"
            id="en"
            onClick={[Function]}
            type="button"
          >
            en
          </button>,
          "className": "col-xs",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "en",
            "className": "primary",
            "id": "en",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "en",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs"
          >
            <button
              className="primary"
              id="pt"
              onClick={[Function]}
              type="button"
            >
              pt
            </button>
          </div>,
          <div
            className="col-xs"
          >
            <button
              className="primary"
              id="en"
              onClick={[Function]}
              type="button"
            >
              en
            </button>
          </div>,
        ],
        "className": "language_selector__languages row center-xs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "pt-BR",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="primary"
              id="pt"
              onClick={[Function]}
              type="button"
            >
              pt
            </button>,
            "className": "col-xs",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "pt",
              "className": "primary",
              "id": "pt",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "pt",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "en-US",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="primary"
              id="en"
              onClick={[Function]}
              type="button"
            >
              en
            </button>,
            "className": "col-xs",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "en",
              "className": "primary",
              "id": "en",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "en",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
